File: hackerrank/30Days/Day10.c
/*Count max number of consecutive 1's in binary representation
of a number*/
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

int main(){
    int n; 
    scanf("%d",&n);
    int bin[100];
    int i=0;
    while(n!=0){
        bin[i++] = n%2;
        n /= 2;
    }
    int maxcount = 0,count=0;
    for(int j=0;j<i;j++){
        if(bin[j]){
            count++;
            if(count>maxcount)
                maxcount=count;
        }
        else
            count=0;
    }
    
    printf("%d",maxcount);
    return 0;
}

File: hackerrank/30Days/Day10.py
#count max number of consecutive 1's in binary repr of a number
#!/bin/python
import sys
n = int(raw_input().strip())
s = ""
i = n
while i:
    if i%2:
        s = "1" + s
    else:
        s = "0" + s
    i/=2
c = 0
d = 0
for t in s:
    if t=="1":
        d+=1
        if d>c:
            c = d
    else:
        d=0
print c

